src/pages/Transfer/CrossChainTransferForm/index.tsx

```diff
@@ -1,52 +1,187 @@
+import STATUSES from 'utils/constants/statuses';
+import { xcmTransfer } from 'utils/relayChainApi/transfer';
+import { newMonetaryAmount } from '@interlay/interbtc-api';
+
+const TRANSFER_AMOUNT = 'transfer-amount';
+
+type CrossChainTransferFormData = {
+  [TRANSFER_AMOUNT]: string;
+}
 
 const CrossChainTransferForm = (): JSX.Element => {
+  const [destination, setDestination] = React.useState<InjectedAccountWithMeta | undefined>(undefined);
+  const [submitStatus, setSubmitStatus] = React.useState(STATUSES.IDLE);
+  const [submitError, setSubmitError] = React.useState<Error | null>(null);
+  const [approxUsdValue, setApproxUsdValue] = React.useState<string>('0');
+
+  const dispatch = useDispatch();
   const { t } = useTranslation();
 
   const {
-    collateralTokenTransferableBalance
+    register,
+    handleSubmit,
+    formState: { errors },
+    reset
+  } = useForm<CrossChainTransferFormData>({
+    mode: 'onChange'
+  });
+
+  const {
+    parachainStatus,
+    address,
+    prices
   } = useSelector((state: StoreType) => state.general);
 
+  const onSubmit = async (data: CrossChainTransferFormData) => {
+    if (!address) return;
+    if (!destination) return;
+
+    try {
+      setSubmitStatus(STATUSES.PENDING);
+
+      await xcmTransfer(
+        address,
+        destination.address,
+        newMonetaryAmount(data[TRANSFER_AMOUNT], COLLATERAL_TOKEN, true)
+      );
+
+      setSubmitStatus(STATUSES.RESOLVED);
+    } catch (error) {
+      setSubmitStatus(STATUSES.REJECTED);
+      setSubmitError(error);
+    }
+  };
+
+  const handleConfirmClick = (event: React.MouseEvent<HTMLButtonElement>) => {
+    if (!address) {
+      dispatch(showAccountModalAction(true));
+      event.preventDefault();
+    }
+  };
+
+  // This ensures that triggering the notification and clearing
+  // the form happen at the same time.
+  React.useEffect(() => {
+    if (submitStatus !== STATUSES.RESOLVED) return;
+
+    toast.success(t('transfer_page.successfully_transferred'));
+
+    reset({
+      [TRANSFER_AMOUNT]: ''
+    });
+  }, [
+    submitStatus,
+    reset,
+    t
+  ]);
+
+  const handleValueChange = (event: any) => {
+    const value = newMonetaryAmount(event.target.value, COLLATERAL_TOKEN, true);
+    const usd = getUsdAmount(value, prices.governanceToken.usd);
+
+    setApproxUsdValue(usd);
+  };
+
   return (
-    <form className='space-y-8'>
-      <FormTitle>
-        {t('transfer_page.cross_chain_transfer_form.title')}
-      </FormTitle>
-      <div>
-        <AvailableBalanceUI
-          label={t('transfer_page.cross_chain_transfer_form.balance')}
-          balance={displayMonetaryAmount(collateralTokenTransferableBalance)} />
+    <>
+      <form
+        className='space-y-8'
+        onSubmit={handleSubmit(onSubmit)}>
+        <FormTitle>
+          {t('transfer_page.cross_chain_transfer_form.title')}
+        </FormTitle>
+        <div>
+          <div>
+            <TokenField
+              onChange={handleValueChange}
```

Comment:
@tomjeatt 
We are using `react-hook-form` which is based on uncontrolled component mode. But here `onChange` is used again which is based on controlled component mode. That's why `setApproxUsdValue` is introduced, which is I think is not efficient. We can use `getValue` or `watch` from `react-hook-form`. Then the code would be simpler and less renderable.

Reply:
OK, that sounds good for now. I hope we don't forget to handle this at that time. :)